src/main/java/Parser.java

```diff
@@ -0,0 +1,189 @@
+
+            Task modifiedTask = tasks.getTask(taskIndex);
+            modifiedTask.markAsDone();
+
+            //Saving
+            fileHandler.saveTasks(tasks);
+
+            //Append to result
+            result.add("Nice! I've marked this task as done:");
+            result.add("  " + modifiedTask.toString());
+
+            break;
+        case "todo":
+            if (inputArray.length < 2) {
+                throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+            }
+            newTask = new Todo(inputArray[1]);
+            tasks.addTask(newTask);
+
+            //Saving
+            fileHandler.saveTasks(tasks);
+
+            //Append to result
+            result.add("Got it. I've added this task:");
+            result.add("  " + newTask.toString());
+            result.add("Now you have " + tasks.getTaskListLength() + " task(s) in the list.");
+
+            break;
+        case "event":
+            if (inputArray.length < 2) {
+                throw new DukeException("OOPS!!! The description of a event cannot be empty.");
+            }
+
+            cmdArgs = inputArray[1].split(" /at ", 2);
+            newTask = new Event(cmdArgs[0], cmdArgs[1]);
+
+            tasks.addTask(newTask);
+
+            //Saving
+            fileHandler.saveTasks(tasks);
+
+            //Append to result
+            result.add("Got it. I've added this task:");
+            result.add("  " + newTask.toString());
+            result.add("Now you have " + tasks.getTaskListLength() + " task(s) in the list.");
+
+            break;
+        case "deadline":
+            if (inputArray.length < 2) {
+                throw new DukeException("OOPS!!! The description of a deadline cannot be empty.");
+            }
+            cmdArgs = inputArray[1].split(" /by ", 2);
+            newTask = new Deadline(cmdArgs[0], cmdArgs[1]);
+
+            tasks.addTask(newTask);
+
+            //Saving
+            fileHandler.saveTasks(tasks);
+
+            //Append to result
+            result.add("Got it. I've added this task:");
+            result.add("  " + newTask.toString());
+            result.add("Now you have " + tasks.getTaskListLength() + " task(s) in the list.");
+
+            break;
+        case "delete":
+            if (inputArray.length < 2) {
+                throw new DukeException("OOPS!!! You need to specify the number of the task " 
+                        + "you want to delete!");
+            }
+            int taskIndexToRemove = Integer.parseInt(inputArray[1]) - 1;
+            Task removedTask = tasks.removeTask(taskIndexToRemove);
+
+            //Saving
+            fileHandler.saveTasks(tasks);
+
+            //Append to result
+            result.add("Noted. I've removed this task:");
+            result.add("  " + removedTask.toString());
+            result.add("Now you have " + tasks.getTaskListLength() + " task(s) in the list.");
+
+            break;
+        case "remind":

```

Comment:
Would it be a better idea to extract all these codes into a separate method? It looks super lengthy and affects readability. This code doesn't fulfil Single Responsibility Principle as it has several reasons to change. Let's say the execution in case "remind" gets changed, your code in this parseAndExecute would have to change (because the remind case must change); then when execution of delete changes, this method would also have reason to change. It increases the chances of having bugs in this method and unit test for this method will be complicated. Instead, my approach is to have handleRemind() and handleDelete() etc. to handle all these executions. These methods can be tested separately and parseAndExecute would have only 1 reason to change: that is, when there are more types of command/execution to be added (say, listFull) and only that.